File: src/main/kotlin/top/bearingwall/asya/model/TimeAnchor.kt
package top.bearingwall.asya.model

import jakarta.persistence.*
import top.bearingwall.asya.util.LocalDateTimeStringConverter
import java.math.BigDecimal
import java.time.LocalDateTime

@Entity
@Table(name = "time_anchors")
class TimeAnchor(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long? = null,

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "conference_id", nullable = false)
    var conference: Conference? = null,

    @Column(name = "update_time")
    var updateTime: LocalDateTime? = null,

    @Column(name = "anchor_real_time")
    var anchorRealTime: LocalDateTime? = null,

    @Column(name = "anchor_game_time")
    @Convert(converter = LocalDateTimeStringConverter::class)
    var anchorGameTime: LocalDateTime? = null,

    @Column(name = "time_ratio", precision = 10, scale = 2)
    var timeRatio: BigDecimal? = null,

    @Column(name = "is_current")
    var isCurrent: Boolean = false
)

File: backend-dotnet/src/AsyaMun.Api/Models/Core/TimeAnchor.cs
namespace AsyaMun.Api.Models;

public class TimeAnchor
{
    public long Id { get; set; }

    public string? AnchorGameTime { get; set; }

    public DateTime? AnchorRealTime { get; set; }

    public bool IsCurrent { get; set; }

    public decimal? TimeRatio { get; set; }

    public DateTime? UpdateTime { get; set; }

    public Guid ConferenceId { get; set; }

    public virtual Conference Conference { get; set; } = null!;
}